### DISCUSSION TITLE
Adding S3 Tags to Archived Images

### DISCUSSION THREAD

**Discussion by RichardLe** [guardian/grid#3382] [Ideas]
The BBC needs to keep an archive copy of images that our archivists decide to keep.  We'd like to identify the images to be sent to the archive using S3 tags.   The use case looks something like:

Given an image that already exists in the grid catalogue
When a user selects "Add to Library" 
Then the image object stored in the main image Bucket should have a tag added
And the key for the tag should be "archive"
And the value for the tag should be "true"

We can either implement this as part of the setArchived endpoint in the metadata service, or as a periodic lambda which queries the api service for archived images and sets the flag. 

The first option makes this part of the Grid's capabilities for all implementers and is our preferred option.  

Would welcome feedback from the Guardian as to whether this is a generally useful feature that should be implemented in the core funcitonality of the Grid.

**Comment by honorcb** [comment]
What Happens when the image has already been added to the Library ( edited metadata,  added label,  cropped or downloaded )  how do you select to Archive this image,   as these actions remove the "Add to Library" option for an image.
